fuel/transformers/__init__.py

```diff
@@ -928,3 +928,50 @@ def __init__(self, data_stream, sources, **kwargs):
     def transform_any(self, data):
         return [d for d, s in izip(data, self.data_stream.sources)
                 if s in self.sources]
+
+
+class OneHotEncoding(SourcewiseTransformer):
+    """Converts integer target variables to one hot encoding.
+
+    It assumes that the targets are integer numbers from 0,... , N-1.
+    Since it works on the fly the number of classes N needs to be
+    specified. The targets are assumed to be in source_name 'targets',
+    but can be specified otherwise.
+
+    Parameters
+    ----------
+    data_stream : :class:`DataStream` or :class:`Transformer`.
+        The data stream.
+    num_classes : int
+        The number of classes.
+    source_name : str, default 'targets'
+        The name of the source that will be transformed.
+
+    """
+    def __init__(self, data_stream, num_classes, source_name='targets',
+                 **kwargs):
+        self.num_classes = num_classes
+        self.source_name = source_name
+
+        super(OneHotEncoding, self).__init__(
+            data_stream, data_stream.produces_examples, **kwargs)
+
+    def transform_source_example(self, source_example, source_name):
+        if source_name == self.source_name:
+            assert source_example <= self.num_classes
```

Comment:
Instead of an assertion, it'd be preferable to raise an informative `ValueError`.